lib/smart_answer/calculators/pay_leave_for_parents_calculator.rb

```diff
@@ -97,60 +97,36 @@ def earnings_employment(earnings_employment, work_employment)
       earnings_employment == "yes" && work_employment == "yes"
     end
 
-    def range_in_2013_2014_fin_year?
-      date_in_39_week_range?(2013, 2014, due_date)
-    end
-
-    def range_in_2014_2015_fin_year?
-      date_in_39_week_range?(2014, 2015, due_date)
-    end
-
-    def range_in_2015_2016_fin_year?
-      date_in_39_week_range?(2015, 2016, due_date)
-    end
-
-    def range_in_2016_2017_fin_year?
-      date_in_39_week_range?(2016, 2017, due_date)
-    end
-
-    def range_in_2017_2018_fin_year?
-      date_in_39_week_range?(2017, 2018, due_date)
+    (2013..2017).each do |year_start|
+      year_end = year_start + 1
+      define_method "range_in_#{year_start}_#{year_end}_fin_year?" do
+        date_in_39_week_range?(year_start, due_date)
+      end
     end
 
     def start_of_maternity_allowance
       sunday_before(due_date - 11.weeks)
     end
 
     def earliest_start_mat_leave
-      sunday_before(due_date - 11.weeks)
+      start_of_maternity_allowance
     end
 
     def maternity_leave_notice_date
       saturday_before(due_date - 14.weeks)
     end
 
     def paternity_leave_notice_date
-      saturday_before(due_date - 14.weeks)
-    end
-
-    def in_2013_2014_fin_year?(date)
-      (Date.new(2013, 05, 06)..Date.new(2014, 05, 05)).cover?(date)
-    end
-
-    def in_2014_2015_fin_year?(date)
-      (Date.new(2014, 05, 06)..Date.new(2015, 05, 05)).cover?(date)
+      maternity_leave_notice_date
```

Comment:
Maybe we could use an alias so that is more clear that these two methods are the same?

```rb
def maternity_leave_notice_date
  saturday_before(due_date - 14.weeks)
end

alias_method :paternity_leave_notice_date :maternity_leave_notice_date
```

Reply:
@andrewgarner Yes that was my first idea. But then I realised that the fact that these methods are calculated in the same way, doesn't mean they are the same. Indeed they are probably not, as the paternity leave and maternity leave could be calculated differently in the future, and I think that defining one them as an alias could be more confusing.

Reply:
I think an alias is a good design decision because it makes it very clear that they are currently the same. If someone in the future comes to change the implementation of `maternity_leave_notice_date` it will be clear to them that they are also changing the implementation of `paternity_leave_notice_date` and can consider if that is the right thing to do at the time.

With the implementation we have now, it might be less clear to future authors that changing the implementation of `maternity_leave_notice_date` has a side effect on `paternity_leave_notice_date`.

Reply:
@andrewgarner ok I agree, I updated it